tests/device_tests/test_protection_levels.py

```diff
@@ -32,239 +33,362 @@
 PIN4 = "1234"
 
 
-@pytest.mark.skip_t2
-class TestProtectionLevels:
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_initialize(self, client):
-        with client:
-            client.set_expected_responses([messages.Features])
-            client.init_device()
-
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_apply_settings(self, client):
-        with client:
-            client.use_pin_sequence([PIN4])
-            client.set_expected_responses(
-                [
-                    messages.PinMatrixRequest,
-                    messages.ButtonRequest,
-                    messages.Success,
-                    messages.Features,
-                ]
-            )  # TrezorClient reinitializes device
-            device.apply_settings(client, label="nazdar")
-
-    @pytest.mark.setup_client(pin=PIN4, passphrase=True)
-    def test_change_pin(self, client):
-        with client:
+pytestmark = pytest.mark.setup_client(pin=PIN4, passphrase=True)
+
+# General NOTE: we need to assert the value of client.features.passphrase_protection
+# at the end, because in case of TT, this value gets filled correcly only after
+# unlocking the device with the PIN.
+# Also, it may be needed to update the client info by client.refresh_features().
+# There are cases when it is even impossible to check the correct TT passphrase_protection,
+# in that case we check it only on T1.
+
+
+def _pin_request(client):
+    """Get appropriate PIN request for each model"""
+    if client.features.model == "1":
+        return messages.PinMatrixRequest
+    else:
+        return messages.ButtonRequest(code=B.PinEntry)
+
+
+def test_initialize(client):
+    with client:
+        assert client.features.pin_protection is True
+        if client.features.model == "1":
+            assert client.features.passphrase_protection is True
+        client.set_expected_responses([messages.Features])
+        client.init_device()
+
+
+@pytest.mark.setup_client(pin=PIN4, passphrase=True)
+def test_apply_settings(client):
+    with client:
+        assert client.features.pin_protection is True
+        client.use_pin_sequence([PIN4])
+        client.set_expected_responses(
+            [
+                _pin_request(client),
+                messages.ButtonRequest,
+                messages.Success,
+                messages.Features,
+            ]
+        )  # TrezorClient reinitializes device
+        device.apply_settings(client, label="nazdar")
+        assert client.features.passphrase_protection is True
+
+
+def test_change_pin(client):
+    with client:
+        assert client.features.pin_protection is True
+        # Models differ both in PIN sequence (TT requires one more input)
```

Comment:
that suggests that this should be two tests

Reply:
Fixed in 699ae28